pkg/artifactory/resource_artifactory_ldap_setting.go

```diff
@@ -0,0 +1,312 @@
+package artifactory
+
+import (
+	"context"
+	"encoding/xml"
+	"github.com/go-resty/resty/v2"
+	"gopkg.in/ldap.v2"
+	"gopkg.in/yaml.v2"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+type LdapSetting struct {
+	Key                      string         `xml:"key" yaml:"key"`
+	Enabled                  bool           `xml:"enabled" yaml:"enabled"`
+	LdapUrl                  string         `xml:"ldapUrl" yaml:"ldapUrl"`
+	UserDnPattern            string         `xml:"userDnPattern" yaml:"userDnPattern"`
+	EmailAttribute           string         `xml:"emailAttribute" yaml:"emailAttribute"`
+	AutoCreateUser           bool           `xml:"autoCreateUser" yaml:"autoCreateUser"`
+	LdapPoisoningProtection  bool           `xml:"ldapPoisoningProtection" yaml:"ldapPoisoningProtection"`
+	AllowUserToAccessProfile bool           `xml:"allowUserToAccessProfile" yaml:"allowUserToAccessProfile"`
+	PagingSupportEnabled     bool           `xml:"pagingSupportEnabled" yaml:"pagingSupportEnabled"`
+	Search                   LdapSearchType `xml:"search" yaml:"search"`
+}
+
+type LdapSearchType struct {
+	SearchSubTree   bool   `xml:"searchSubTree" yaml:"searchSubTree" `
+	SearchFilter    string `xml:"searchFilter" yaml:"searchFilter"`
+	SearchBase      string `xml:"searchBase" yaml:"searchBase"`
+	ManagerDn       string `xml:"managerDn" yaml:"managerDn"`
+	ManagerPassword string `xml:"managerPassword" yaml:"managerPassword"`
+}
+
+type LdapSettings struct {
+	LdapSettingArr []LdapSetting `xml:"ldapSetting" yaml:"ldapSetting"`
+}
+
+type SecurityLdapSettings struct {
+	LdapSettings LdapSettings `xml:"ldapSettings"`
+}
+
+type XmlLdapConfig struct {
+	XMLName  xml.Name             `xml:"config"`
+	Security SecurityLdapSettings `xml:"security"`
+}
+
+func resourceArtifactoryLdapSetting() *schema.Resource {
+	var ldapSettingsSchema = map[string]*schema.Schema{
+		"key": {
+			Type:             schema.TypeString,
+			Required:         true,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotEmpty),
+			Description:      `(Required) Ldap setting name.`,
+		},
+		"enabled": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     true,
+			Description: `(Optional) Flag to enable or disable the ldap setting. Default value is "true".`,
+		},
+		"ldap_url": {
+			Type:             schema.TypeString,
+			Required:         true,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.IsURLWithScheme([]string{"ldap", "ldaps"})),
+			Description:      "(Required) Location of the LDAP server in the following format: ldap://myldapserver/dc=sampledomain,dc=com",
+		},
+		"user_dn_pattern": {
+			Type:     schema.TypeString,
+			Required: true,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.All(
```

Comment:
You've repeated this code - put this in the same place as `validateIsEmail` and reuse it

Reply:
made new functions in validators.go and reused them.